File: clinical/src/test/java/com/hartwig/actin/clinical/feed/vitalfunction/VitalFunctionExtractionTest.java
package com.hartwig.actin.clinical.feed.vitalfunction;

import static org.junit.Assert.assertEquals;

import com.hartwig.actin.clinical.datamodel.VitalFunctionCategory;

import org.junit.Test;

public class VitalFunctionExtractionTest {

    @Test
    public void canDetermineVitalFunctionCategory() {
        assertEquals(VitalFunctionCategory.NON_INVASIVE_BLOOD_PRESSURE, VitalFunctionExtraction.determineCategory("NIBP"));
        assertEquals(VitalFunctionCategory.ARTERIAL_BLOOD_PRESSURE, VitalFunctionExtraction.determineCategory("ABP"));
        assertEquals(VitalFunctionCategory.HEART_RATE, VitalFunctionExtraction.determineCategory("HR"));
        assertEquals(VitalFunctionCategory.SPO2, VitalFunctionExtraction.determineCategory("SpO2"));
    }

    @Test(expected = IllegalArgumentException.class)
    public void crashOnInvalidVitalFunctionCategory() {
        VitalFunctionExtraction.determineCategory("not a category");
    }
}

File: clinical/src/test/kotlin/com/hartwig/actin/clinical/feed/emc/vitalfunction/VitalFunctionExtractionTest.kt
package com.hartwig.actin.clinical.feed.emc.vitalfunction

import com.hartwig.actin.clinical.feed.emc.vitalfunction.VitalFunctionExtraction.determineCategory
import com.hartwig.actin.datamodel.clinical.VitalFunctionCategory
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class VitalFunctionExtractionTest {

    @Test
    fun `Should determine vital function category`() {
        assertThat(determineCategory("NIBP")).isEqualTo(VitalFunctionCategory.NON_INVASIVE_BLOOD_PRESSURE)
        assertThat(determineCategory("ABP")).isEqualTo(VitalFunctionCategory.ARTERIAL_BLOOD_PRESSURE)
        assertThat(determineCategory("HR")).isEqualTo(VitalFunctionCategory.HEART_RATE)
        assertThat(determineCategory("SpO2")).isEqualTo(VitalFunctionCategory.SPO2)
    }

    @Test(expected = IllegalArgumentException::class)
    fun `Should throw exception on invalid vital function category`() {
        determineCategory("not a category")
    }
}